utility/src/main/java/org/oppia/util/data/UrlImageParser.kt

```diff
@@ -25,21 +34,20 @@ class UrlImageParser(
 
   val GCS_PREFIX: String = "https://storage.googleapis.com/"
```

Comment:
This & the bucket name should be dagger-provided constants since we'll use them elsewhere. Ditto for the image URL template below (which can be formatted as such: ``"%s/%s/assets/image/%s"`` and we can use String.format to fill it in). Please move them to a dagger module (see https://github.com/oppia/oppia-android/blob/develop/utility/src/main/java/org/oppia/util/logging/LoggerModule.kt for a similar example, but with booleans instead of strings).

Reply:
Note that to make this work you probably need to create Factory for UrlImageParser that's injectable, and hide the parser's constructor. The factory can create new instances and pass injected values to it (similar to https://github.com/oppia/oppia-android/blob/develop/data/src/main/java/org/oppia/data/persistence/PersistentCacheStore.kt#L213).

Reply:
@BenHenning I am not able to create Factory for ImageParser. I am not getting clear idea on how to do this.

Reply:
Duplicate of this PR is created #205